### ISSUE TITLE
About the feature norm and the defect of A-Softmax

### ISSUE THREAD

**Issue by gehaocool** [KaleidoZhouYN/FaceRecognitionSystem#1]
Hi, I've been following your repos lately.
In [Loss-Functions](https://github.com/KaleidoZhouYN/Loss-Functions) you've mentioned there's a defect of softmax especially when it's used in sphereface and A-softmax.
Do you mean that when the training is going on, the loss is going down at the same time, but the performance of the net is declining?
And do you think the reason is about feature norm as what you said in this repo [FaceRecognitionSystem](https://github.com/KaleidoZhouYN/FaceRecognitionSystem)?

Actually I‘ve I encountered this situation a lot.
I'm training sphereface-36 on vggdata2, but I removed the identities overlap with lfw.
The m=1 pretrain step is normal, I stop it at 300000 iterations, and test on lfw result 97.27%
Then I turned m=2 and lambda=3 lr=0.01, at first it's OK, the net collapsed at about 180000 iterations, and test on lfw result is 99.48%. 
When I changed lambda to 2 or 1 and continue the training from the model of 180000, the net just can't get better, while the loss is going down. 
Then I test last model, and the 170000, 160000, 150000 models are even better than the former one.
I've tried to reduce lr, change lambda_min, but still the same.
Can you help me with this?
